include/flashinfer/sampling.cuh

```diff
@@ -1175,10 +1178,28 @@ __global__ void ChainSpeculativeSampling(DType* draft_probs, IdType* draft_token
       // accept the draft models output
       output_token_ids[row_idx * (num_speculative_tokens + 1) + pos] = draft_id;
     } else {
+      stop_pos = pos;
       break;
     }
   }
 
+  uint32_t emitted_token_num = stop_pos;
+  uint32_t accepted_token_num = stop_pos;
+  for (uint32_t i = stop_pos; i < num_speculative_tokens; ++i) {
+    IdType draft_id = draft_token_ids[row_idx * num_speculative_tokens + i];
+    float q = target_probs[(row_idx * (num_speculative_tokens + 1) + i) * d + draft_id],
+          p = draft_probs[(row_idx * num_speculative_tokens + i) * d + draft_id];
+    DType u = uniform_samples[row_idx * (num_speculative_tokens + 1) + i];
+    if (u * p < q) {
+      ++accepted_token_num;
+    }
+  }
+
+  if (tx == 0) {
+    output_accepted_token_num[row_idx] = accepted_token_num;
+    output_emitted_token_num[row_idx] = emitted_token_num;
```

Comment:
Can we make it in-place update? If user specify an `output_accepted_token_num` array, we update its value directly, otherwise we return an a new array.

```suggestion
    output_accepted_token_num[row_idx] += accepted_token_num;
    output_emitted_token_num[row_idx] += emitted_token_num;
```